### PR TITLE
Update api for DHL Pickup Request

### PR DESCRIPTION
**Pull request by OlfatProgressus**

### CODE REVIEW

File: pr-dhl-woocommerce/includes/REST_API/Paket/Client.php [hunk 1]
```diff
@@ -253,31 +254,39 @@ protected function request_pickup_info_to_request_data( Pickup_Request_Info $req
...
+				'city'			=> $request_pickup_info->pickup_address['city'],
+				'postalCode' 	=> $request_pickup_info->pickup_address['postalCode'],
+				'state' 		=> $request_pickup_info->pickup_address['state'],
+				'country'		=> $request_pickup_info->pickup_address['country']
 			)
 		);
-		$business_hours = null;
-		if ( $request_pickup_info->business_hours ) {
-			$pickup_location_array['businessHours']	= $request_pickup_info->business_hours;
-		}
```
Comment (Abdalsalaam) [inline pr-dhl-woocommerce/includes/REST_API/Paket/Client.php:264]: Why was this data deleted?

**Review by Abdalsalaam — CHANGES_REQUESTED** [review]

File: pr-dhl-woocommerce/includes/REST_API/Paket/Client.php [hunk 2]
```diff
@@ -253,31 +254,39 @@ protected function request_pickup_info_to_request_data( Pickup_Request_Info $req
...
-				'name' 	=> $request_pickup_info->pickup_contact['name'],
-				'phone' 	=> $request_pickup_info->pickup_contact['phone'],
-				'email' 	=> $request_pickup_info->pickup_contact['email'],
+				array(
+					'name' 			   => $request_pickup_info->pickup_contact['name'],
+					'phone' 		   => $request_pickup_info->pickup_contact['phone'],
+					'email' 		   => $request_pickup_info->pickup_contact['email'],
+					"emailNotification"=> array(
+						"sendPickupConfirmationEmail" => "true",
+						"sendPickupTimeWindowEmail"	  => "true"
```
Comment (Abdalsalaam) [inline pr-dhl-woocommerce/includes/REST_API/Paket/Client.php:287]: And I'm also wondering why you set this to "true"? can you check this with Shadi, is there any setting for this? is it required? I can see it was disabled in the previous code.

File: pr-dhl-woocommerce/includes/REST_API/Paket/Client.php [hunk 3]
```diff
@@ -109,15 +110,15 @@ public function create_pickup_request ( Pickup_Request_Info $pickup_request_info
...
+					$this->generate_error_details( $error_msg . print_r( $response->body, true ) )
 				)
 			);
 		}
 
 		throw new Exception(
 			sprintf(
 				__( 'Failed DHL Request Pickup: %s', 'dhl-for-woocommerce' ),
-				$this->generate_error_details( $response->body )
+				$this->generate_error_details( print_r( $response->body, true ) )
```
[the remark this answers is not in the document]
Comment (OlfatProgressus) [reply pr-dhl-woocommerce/includes/REST_API/Paket/Client.php]: Fixed by target the title $response->body[0]->title

File: pr-dhl-woocommerce/includes/REST_API/Paket/Client.php (hunk 2 above)
Comment (OlfatProgressus) [reply pr-dhl-woocommerce/includes/REST_API/Paket/Client.php]: When tested it with the one used before , it gives error 400 so these values are mandatory

**Review by Abdalsalaam — APPROVED** [review]

**Pull request merged**